game/views.py

```
@@ -432,28 +430,24 @@ def offer_draw(request):
-
+    resigning_player = game.player_color if game.mode == 'ai' else game.current_turn
     winner = 'black' if resigning_player == 'white' else 'white'
-
     game_status = 'resignation'
 
     game.game_status = game_status
     request.session['game'] = game.to_dict()
     request.session.modified = True
 
+try:
     record_game_result(request, game.mode, winner, 'resign', game.player_color)
```

Ruff (0.15.13)

[error] Undefined name `request`

(F821)